### ISSUE TITLE
EET support

### ISSUE THREAD

**Issue by MephistoSatanDevil** [critto-bg/ImprovedAnvil#38]
Is it possible to modify this mod so that it can be installed within EET to enhance the content of the BG2 section within EET?

**Comment by critto-bg** [comment]
I have never tried, but the technical part could probably be resolved with some effort.

The harder problem is game balance. If I remember correctly, you start BG2 part at higher XP after playing through SoD, something like 500k XP? This will throw the whole balance of IA off. The mod was designed ages ago for parties starting at the original vanilla XP level.

**Comment by MephistoSatanDevil** [comment]
Might it be more considerate to prioritize "installation support" initially, and then gently shift our focus to the game's balance issues once the installation process is smoothly completed?

Would it be more better to make the adjustment from "REQUIRE_PREDICATE GAME_IS ~bg2ee~ @8901" to "REQUIRE_PREDICATE GAME_IS ~bg2ee eet~ @8901" to initially address the installation feasibility?
Would it be better to consider the rest at a later time?

**Comment by critto-bg** [comment]
We can of course attempt to make it compatible on a purely technical level (installs well without any issues). I will investigate the subject and, if it is easily achievable, add the support in the near future. Something tells me there's more to adjust than simply allowing the EET variant as a predicate.

But when it comes to game balance, I am not sure if it is at all possible without significant rework.

**Comment by MephistoSatanDevil** [comment]
Hi @critto-bg ,
I have modified a version that supports EET, and the install test on BG2EE and EET are normal. 
However, I am unsure how to quickly check if everything is correctly during the game. Could you please assist me in conducting the checks?
https://github.com/MephistoSatanDevil/ImprovedAnvil

**Comment by MephistoSatanDevil** [comment]
Perhaps the restrictions on experience points could be adjusted as an option, or could some of the battle difficulties be dynamically adjusted based on options?

**Comment by critto-bg** [comment]
Hi! The related commit looks huge, many updates, hard to say if everything is fine. I will try to give it a more detailed look very soon. Some files are shown as completely updated (all lines are changed), what is the actual scope of changes in those files?

As for testing, I'm afraid there isn't really any good way except to do a play-through and squash the bugs that pop up along the way. I have never tried EET myself and therefore not aware if there are any differences that might fundamentally affect the experience. Logically, there shouldn't be, so long as the names of resources and objects stay the same, and the workaround for the chapter numbers that you did works okay. 

The XP issue is another one I don't have a quick solution for, unfortunately. There isn't a centralised system within the mod for manipulating the XP gains on a global level. Pretty much every encounter is hand-crafted based on the original vanilla XP starting point and the supposed order in which a player might tackle that encounter during their game.

**Comment by critto-bg** [comment]
I just saw that your original comment about the modification was almost 1.5 months ago (sorry I missed it, didn't get a notification for some reason). Did you have a chance to do an actual playtest of the changes?

**Comment by MephistoSatanDevil** [comment]
The previous original comment is irrelevant and doesn't need much attention. This modification was done using the tools mentioned in the EET mod modification tutorial I found a couple of days ago. I am an EET player, and therefore not aware if there are any differences that might fundamentally affect the experience.

**Comment by critto-bg** [comment]
So you decided to discard the aforementioned patch altogether?

**Comment by MephistoSatanDevil** [comment]
No, I am also looking for someone familiar with IA to help me with testing.
Also, I believe that even if there are any issues, the patch mentioned above might be help for you.

**Comment by critto-bg** [comment]
Well, since nothing has been changed fundamentally about the logic of the mod, the only things that can cause trouble are EET-specific. I am not familiar with EET to make any educated guesses. So, as I have already mentioned, the only good way to test in this scenario is to make a play-through and fix the bugs that come up.

I definitely do not have any plans to play the game in the foreseeable future, especially considering that support for EET has never been our goal. But if you do consider making a test run, let me know how it goes and what bugs would you encounter (if any), I'll try to advise to the best of my ability how to fix them.

**Comment by MephistoSatanDevil** [comment]
OK.

The first issue we encountered is that when starting a new SoA game, Imoen does not appear, which results in the protagonist not being rescued and the game being unplayable. 
In my previous tests, when transitioning from the SoD to SoA, Imoen's behavior was normal.

**Comment by critto-bg** [comment]
Understood. I suspect I'll have to try and make a setup on my own machine to see this in action.

Does the same thing happen when you start a fresh SoA game on a EET installation?

**Comment by MephistoSatanDevil** [comment]
When I start a fresh SoA game on EET installation without IA, it is completely normal. The problem mentioned above only occurs after installing IA.

**Comment by critto-bg** [comment]
Understood. Looks like it requires some debugging. I will try to dedicate some time to this in the near future.
